Sources/StreamChat/Controllers/ListDatabaseObserver.swift

```diff
@@ -131,7 +131,9 @@ class ListDatabaseObserver<Item, DTO: NSManagedObject> {
             cacheName: nil
         )
         
-        _items.computeValue = { [unowned self] in (self.frc.fetchedObjects ?? []).lazy.compactMap(self.itemCreator) }
+        _items.computeValue = { [unowned self] in
+            LazyCachedMapCollection(source: self.frc.fetchedObjects ?? [], map: self.itemCreator)
+        }
```

Comment:
All collections in `Foundation` can be constructed using some convenience extension method. I think we should follow this practice so the code looks more natural. So maybe something like `.lazyCacheMap { }` 🤔